File: src/javacode/DiagonalDifference.java
package javacode;

import java.io.*;
import java.util.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;


public class DiagonalDifference {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
//        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));
        BufferedWriter bufferedWriter = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<List<Integer>> arr = new ArrayList<>();

        IntStream.range(0, n).forEach(i -> {
            try {
                arr.add(
                    Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                        .map(Integer::parseInt)
                        .collect(toList())
                );
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        });

        int result = diagonalDifference(arr);

        bufferedWriter.write(String.valueOf("\n\n\n\n"+result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }

    public static int diagonalDifference(List<List<Integer>> arr) {
        // Write your code here

        int matrixSize = arr.size();
        int ltrDiagonalSum = 0;
        int rtlDiagonalSum = 0;
        for(int i = 0;i<matrixSize;i++){

            for(int j =0 ;j<matrixSize;j++){
                if(i == j){
                    ltrDiagonalSum += arr.get(i).get(j);
                }
                if(i+j == matrixSize-1){
                    rtlDiagonalSum += arr.get(i).get(j);
                }
            }
        }

        System.out.println(ltrDiagonalSum);
        System.out.println(rtlDiagonalSum);
        return Math.abs(ltrDiagonalSum - rtlDiagonalSum);
    }
}

File: src/kotlincode/DiagonalDifference.kt
import java.io.*
import java.math.*
import java.security.*
import java.text.*
import java.util.*
import java.util.concurrent.*
import java.util.function.*
import java.util.regex.*
import java.util.stream.*
import kotlin.collections.*
import kotlin.comparisons.*
import kotlin.io.*
import kotlin.jvm.*
import kotlin.jvm.functions.*
import kotlin.jvm.internal.*
import kotlin.math.abs
import kotlin.ranges.*
import kotlin.sequences.*
import kotlin.text.*

/*
 * Complete the 'diagonalDifference' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts 2D_INTEGER_ARRAY arr as parameter.
 */

fun diagonalDifference(arr: Array<Array<Int>>): Int {
    // Write your code here
    val arraySize = arr.size
    var ltrDiagonalSum = 0;
    var rtlDiagonalSum = 0;

    for(i in 0 until arraySize){
        for(j in 0 until arraySize){
            if(i == j){
                ltrDiagonalSum += arr[i][j];
            }
        }
    }

    for(i in 0 until arraySize){
        for(j in 0 until arraySize){
            if(i+j == arraySize -1){
                rtlDiagonalSum += arr[i][j];
            }
        }
    }

    return abs(ltrDiagonalSum - rtlDiagonalSum)
}

fun main(args: Array<String>) {
    val n = readLine()!!.trim().toInt()

    val arr = Array<Array<Int>>(n, { Array<Int>(n, { 0 }) })

    for (i in 0 until n) {
        arr[i] = readLine()!!.trimEnd().split(" ").map{ it.toInt() }.toTypedArray()
    }

    val result = diagonalDifference(arr)

    println(result)
}

